### PR TITLE
Sdk hookup

### PR DESCRIPTION
**Pull request by einari**
This provides the hookup for EventHandlers towards runtime and has the start of generic filter and event processor as well. The latter needs more attention, but at least its a start and dabble with the APIs. 

This leverages whats already in this PR: https://github.com/dolittle-fundamentals/DotNET.Fundamentals/pull/289 and what sits in Contracts for the 5.0.0 branch: https://github.com/dolittle-runtime/Contracts/tree/5.0.0

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1140654762299587/1161655544625099)

### CODE REVIEW

File: Source/Events.Handling/EventHandlerMethod.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        /// <inheritdoc/>
+        public async Task Invoke(ICanHandleEvents handler, CommittedEvent @event)
+        {
+            var result = _methodInfo.Invoke(handler, new[] { @event.Event }) as Task;
+            await result.ConfigureAwait(false);
+        }
+
+        void ThrowIfEventHandlerMethodIsNotAsynchronous(MethodInfo methodInfo)
+        {
+            if (!typeof(Task).IsAssignableFrom(methodInfo.ReturnType))
```
Comment (woksin) [inline Source/Events.Handling/EventHandlerMethod.cs:42]: Smells like a Roslyn analyzer

Comment (einari) [reply Source/Events.Handling/EventHandlerMethod.cs:42]: I've added a task for us to do that as well. But we need the additional safety at Runtime.

Comment (einari) [reply Source/Events.Handling/EventHandlerMethod.cs:42]: PS: Its in our Kanban board on the non-synced tasks for Roslyn extensions

File: Source/Events.Processing/EventProcessorAttribute.cs [hunk 2]
```diff
@@ -2,7 +2,6 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using Dolittle.Runtime.Events.Processing;
 
 namespace Dolittle.Events.Processing
 {
```
Comment (woksin) [inline Source/Events.Processing/EventProcessorAttribute.cs:7]: Shouldn't this file be removed now?

Comment (einari) [reply Source/Events.Processing/EventProcessorAttribute.cs:7]: Not yet. We will keep it till we have integrated fully the first customer project.
In other words, 3 more weeks :)

File: Source/Events.Processing/PartitionId.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+{
+    /// <summary>
+    /// Represents the concept of a unique identifier for a partition.
+    /// </summary>
+    public class PartitionId : ConceptAs<Guid>
+    {
+        /// <summary>
+        /// The identifier for when a partition is not specified.
+        /// </summary>
+        public static PartitionId Unspecificied = Guid.Empty;
```
Comment (woksin) [inline Source/Events.Processing/PartitionId.cs:17]: This is called NotSet in Runtime (I think?) Does it make sense to have them synced or not? If not, what should we go for?

Comment (einari) [reply Source/Events.Processing/PartitionId.cs:17]: Consistency is good. I'm not married to either - I guess we've set NotSet for most other things. Reason I used Unspecified is that its conveying a different meaning - we're not really setting the Partition, but specifying how to partition.

**Review by woksin — APPROVED** [review]

File: Source/Events.Processing/PartitionId.cs (hunk 3 above)
Comment (woksin) [reply Source/Events.Processing/PartitionId.cs:17]: I also liked Unspecified

**Pull request merged**